.github/workflows/step_test.yaml

```diff
@@ -0,0 +1,86 @@
+name: test
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  tests:
+    name: Check on ${{ matrix.toolchain }} toolchain ${{ matrix.mpi }}
+    runs-on: ${{ matrix.os || 'ubuntu-latest' }}
```

Comment:
There is no `macos` and `windows` here, because we need to discuss what compilers to use there and where to get them